packages/cli/src/runners/__tests__/task-broker.test.ts

```diff
@@ -0,0 +1,284 @@
+			};
+
+			const expiredOffer1: TaskOffer = {
+				offerId: 'expired1',
+				runnerId: 'runner2',
+				taskType: 'taskType1',
+				validFor: 1000,
+				validUntil: now - BigInt(1000 * 1_000_000), // 1 second in the past
+			};
+
+			const expiredOffer2: TaskOffer = {
+				offerId: 'expired2',
+				runnerId: 'runner3',
+				taskType: 'taskType1',
+				validFor: 2000,
+				validUntil: now - BigInt(2000 * 1_000_000), // 2 seconds in the past
+			};
+
+			taskBroker.setPendingTaskOffers([validOffer, expiredOffer1, expiredOffer2]);
+
+			taskBroker.expireTasks();
+
+			const offers = taskBroker.getPendingTaskOffers();
+
+			expect(offers).toHaveLength(1);
+			expect(offers[0]).toEqual(validOffer);
+		});
+	});
+
+	describe('registerRunner', () => {
+		it('should add a runner to known runners', () => {
+			const runnerId = 'runner1';
+			const runner = mock<TaskRunner>({ id: runnerId });
+			const messageCallback = jest.fn();
+
+			taskBroker.registerRunner(runner, messageCallback);
+
+			const knownRunners = taskBroker.getKnownRunners();
+			const runnerIds = Object.keys(knownRunners);
+
+			expect(runnerIds).toHaveLength(1);
+			expect(runnerIds[0]).toEqual(runnerId);
+
+			expect(knownRunners[runnerId].runner).toEqual(runner);
+			expect(knownRunners[runnerId].messageCallback).toEqual(messageCallback);
+		});
+	});
+
+	describe('registerRequester', () => {
+		it('should add a requester to known requesters', () => {
+			const requesterId = 'requester1';
+			const messageCallback = jest.fn();
+
+			taskBroker.registerRequester(requesterId, messageCallback);
+
+			const knownRequesters = taskBroker.getKnownRequesters();
+			const requesterIds = Object.keys(knownRequesters);
+
+			expect(requesterIds).toHaveLength(1);
+			expect(requesterIds[0]).toEqual(requesterId);
+
+			expect(knownRequesters[requesterId]).toEqual(messageCallback);
+		});
+	});
+
+	describe('deregisterRunner', () => {
+		it('should remove a runner from known runners', () => {
+			const runnerId = 'runner1';
+			const runner = mock<TaskRunner>({ id: runnerId });
+			const messageCallback = jest.fn();
+
+			taskBroker.registerRunner(runner, messageCallback);
+			taskBroker.deregisterRunner(runnerId);
+
+			const knownRunners = taskBroker.getKnownRunners();
+			const runnerIds = Object.keys(knownRunners);
+
+			expect(runnerIds).toHaveLength(0);
+		});
+	});
+
+	describe('deregisterRequester', () => {
+		it('should remove a requester from known requesters', () => {
+			const requesterId = 'requester1';
+			const messageCallback = jest.fn();
+
+			taskBroker.registerRequester(requesterId, messageCallback);
+			taskBroker.deregisterRequester(requesterId);
+
+			const knownRequesters = taskBroker.getKnownRequesters();
+			const requesterIds = Object.keys(knownRequesters);
+
+			expect(requesterIds).toHaveLength(0);
+		});
+	});
+
+	describe('settleTasks', () => {
+		it('should match task offers with task requests by task type', () => {
```

Comment:
Also, need to test when a new offer or request is added, not just when many of them already exist, since it's actually the least likely scenario.